src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,170 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkMiddle(value, getNodeAtIndex(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIsElementIndex(index);
+        return getNodeAtIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIsElementIndex(index);
+        Node<T> current = getNodeAtIndex(index);
+        T oldValue = current.item;
+        current.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIsElementIndex(index);
+        return unlink(getNodeAtIndex(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> current = first; current != null; current = current.next) {
+            if (current.item == object || current.item != null && current.item.equals(object)) {
+                unlink(current);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private T unlink(Node<T> node) {
+        final T element = node.item;
+        Node<T> next = node.next;
+        Node<T> prev = node.prev;
+
+        if (prev == null) {
+            first = next;
+        } else {
+            prev.next = next;
+            node.prev = null;
+        }
+        if (next == null) {
+            last = prev;
+        } else {
+            next.prev = prev;
+            node.next = null;
+        }
+        node.item = null;
+        size--;
+        return element;
+    }
+
+    private void linkMiddle(T element, Node<T> currentIndexNode) {
+        Node<T> prevNode = currentIndexNode.prev;
+        Node<T> newNode = new Node<>(prevNode, element, currentIndexNode);
+        currentIndexNode.prev = newNode;
+        if (prevNode == null) {
+            first = newNode;
+        } else {
+            prevNode.next = newNode;
+        }
```

Comment:
prevNode never will be a null, cause you always set a newNode there
```suggestion
        if (prevNode == null) {
            first = newNode;
        } else {
            prevNode.next = newNode;
        }
         currentIndexNode.prev = newNode;
```